sunpy/net/dataretriever/sources/goes.py

```diff
@@ -58,18 +58,28 @@ def _get_url_for_timerange(self, timerange, **kwargs):
             Data type to return for the particular GOES satellite. Supported
             types depend on the satellite number specified. (default = xrs_2s)
         """
-
+        start_date = parse_time(timerange.start)
+        end_date = parse_time(timerange.end)
+        
         # find out which satellite and datatype to query from the query times
         sat_num = self._get_goes_sat_num(timerange.start, timerange.end)
         base_url = 'http://umbra.nascom.nasa.gov/goes/fits/'
 
-        if timerange.start < parse_time('1999/01/15'):
-            url = base_url + "{date:%Y}/go{sat:02d}{date:%y%m%d}.fits".format(
-                date=timerange.start, sat=sat_num[0])
-        else:
-            url = base_url + "{date:%Y}/go{sat:02d}{date:%Y%m%d}.fits".format(
-                date=timerange.start, sat=sat_num[0])
-        return [url]
+        total_days = (end_date - start_date).days + 1
```

Comment:
using the `timerange` you can get the number of days, so no need to run this from the converted dates.
`timerange.days`.
The `+1` I imagine is to account for the last day in the range, but I think we had somewhere agree to have a close start and an open end: [tstart, tend); so the end date is not included... but maybe @cadair remembers better